Consolidated tokenization logic and added documentation details (#64)
cisco/mindmeld: source/userguide/nlp.rst

and ISO3166 alpha 2 country code separated by an underscore character, for example, `en_US` and ISO 639-2 language codes, for example, `EN` are used here. We provide a code snippet of how to use this functionality below, assuming you have
spanish training data for such queries:

.. code:: python

    nlp.process('despiértame a las 8', language='ES')

.. code-block:: console

       { 'domain': 'times_and_dates',
         'entities': [ { 'role': None,
                         'span': {'end': 19, 'start': 18},
                         'text': '8',
                         'type': 'sys_time',
                         'value': [ { 'grain': 'hour',
                                      'value': '2019-02-16T08:00:00.000-08:00'}]}],
         'intent': 'set_alarm',
         'text': 'despiértame a las 8'
       }

.. note::

   If you are using Duckling for system entity resolution (the default), the following non-english languages are only supported: `ducking languages <https://github.com/facebook/duckling/blob/master/Duckling/Locale.hs>`_

.. _stemming:

Language stemming
^^^^^^^^^^^^^^^^^

> Stemming is an important, language-dependent NLP processing that transforms a word to it's root word. MindMeld allows you to specify their own custom stemmers,

Reviewer: ```suggestion
Stemming is an important, language-dependent NLP process that transforms a word to an approximation of its root form. MindMeld allows you to specify your own custom stemmers,
```

Author: resolved
